src/mgmt/srv.c

```diff
@@ -389,228 +505,175 @@ process_createpool_request(Drpc__Call *drpc_req,
+		mgmt__destroy_pool_req__free_unpacked(req, NULL);
+		return;
+	}
 
-	rc = uuid_parse(daos_req->uuid, uuid);
+	/* Response status is populated with SUCCESS on init. */
+	mgmt__destroy_pool_resp__init(resp);
+
+	rc = uuid_parse(req->uuid, uuid);
 	if (rc != 0) {
-		D_ERROR("Unable to parse pool UUID %s: %d\n", daos_req->uuid,
+		D_ERROR("Unable to parse pool UUID %s: %d\n", req->uuid,
 			rc);
 		goto out;
 	}
 
 	/* Sys and force params are currently ignored in receiver. */
-	rc = ds_mgmt_destroy_pool(uuid, daos_req->sys,
-				  (daos_req->force == true) ? 1 : 0);
+	rc = ds_mgmt_destroy_pool(uuid, req->sys,
+				  (req->force == true) ? 1 : 0);
 	if (rc != 0) {
-		D_ERROR("Failed to destroy pool %s: %d\n", daos_req->uuid, rc);
+		D_ERROR("Failed to destroy pool %s: %d\n", req->uuid, rc);
 		goto out;
 	}
-out:
-	mgmt__destroy_pool_req__free_unpacked(daos_req, NULL);
-	if (rc != 0)
-		daos_resp->status = MGMT__DAOS_REQUEST_STATUS__ERR_UNKNOWN;
-}
-
-static void
-process_setup_request(Drpc__Call *drpc_req, Mgmt__DaosResp *daos_resp)
-{
-	/* response status is populated with SUCCESS on init */
-	mgmt__daos_resp__init(daos_resp);
 
-	D_DEBUG(DB_MGMT, "Received request to set up server\n");
-
-	dss_init_state_set(DSS_INIT_STATE_SET_UP);
-}
-
-static void
-pack_daos_response(Mgmt__DaosResp *daos_resp, Drpc__Response *drpc_resp)
-{
-	uint8_t	*body;
-	size_t	len;
-
-	len = mgmt__daos_resp__get_packed_size(daos_resp);
+out:
+	resp->status = rc;
+	len = mgmt__destroy_pool_resp__get_packed_size(resp);
 	D_ALLOC(body, len);
 	if (body == NULL) {
 		drpc_resp->status = DRPC__STATUS__FAILURE;
 		D_ERROR("Failed to allocate drpc response body\n");
-		return;
-	}
-
-	if (mgmt__daos_resp__pack(daos_resp, body) != len) {
-		drpc_resp->status = DRPC__STATUS__FAILURE;
-		D_ERROR("Unexpected num bytes for daos resp\n");
-		return;
+	} else {
+		mgmt__destroy_pool_resp__pack(resp, body);
+		drpc_resp->body.len = len;
+		drpc_resp->body.data = body;
 	}
 
-	/* Populate drpc response body with packed daos response */
-	drpc_resp->body.len = len;
-	drpc_resp->body.data = body;
+	mgmt__destroy_pool_req__free_unpacked(req, NULL);
+	D_FREE(resp);
 }
 
 static void
-process_drpc_request(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
+process_setup_request(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
 {
-	Mgmt__DaosResp		*daos_resp = NULL;
-	Mgmt__JoinResp		*join_resp;
-	Mgmt__GetAttachInfoResp	*getattachinfo_resp;
-	Mgmt__CreatePoolResp	*create_pool_resp;
-	uint8_t			*body;
-	size_t			len;
+	Mgmt__DaosResp	*resp = NULL;
+
+	D_DEBUG(DB_MGMT, "Received request to start MS\n");
 
-	D_ALLOC_PTR(daos_resp);
-	if (daos_resp == NULL) {
+	D_ALLOC_PTR(resp);
+	if (resp == NULL) {
 		drpc_resp->status = DRPC__STATUS__FAILURE;
 		D_ERROR("Failed to allocate daos response ref\n");
-
 		return;
 	}
 
+	/* Response status is populated with SUCCESS on init. */
+	mgmt__daos_resp__init(resp);
+
+	D_DEBUG(DB_MGMT, "Received request to set up server\n");
```

Comment:
Redundant message? Above it says "Received request to start MS"

Reply:
https://github.com/daos-stack/daos/pull/816